### PR TITLE
Message inbox and chat interface ui fixed

### PR DESCRIPTION
**Pull request by BHS-Harish**
**Issue #201 fixed**

**Once the issue #208 to be fixed , It will works fine**

- [x] Responsiveness checked

**Screenshots**
InboxUI
image

Chat UI
image

### CODE REVIEW

File: frontend/controller/connectionsController.js [hunk 1]
```diff
@@ -208,7 +208,7 @@ const getAllConversations = async (userId, userType, setConversations, showAlert
       const conversations = response.data.conversations.map(conversation => {
         // Find the participant that is not the current user
         const participant = conversation.participants.find(p => p._id !== userId);
-        const lastMessage = conversation.messages.length > 0 ? conversation.messages[conversation.messages.length - 1] : null;
+        const lastMessage = conversation.messages.length > 0 ? conversation.messages[0] : null;
```
Comment (Kajol-Kumari) [inline frontend/controller/connectionsController.js:211]: why this change? last message should be `messages[conversation.messages.length - 1]` no?

File: frontend/screens/chat/ChatInterface.js [hunk 2]
```diff
@@ -7,79 +7,93 @@ import {
...
-                      <Text style={[styles.timeAgo, styles.timeText]}>
-                        {message.timeAgo}
-                      </Text>
-                    </View>
-                  </View>
-                </View>
-              </View>
+              message.sender.name=="You"?
+              <SenderMessage key={index} name={message.sender.name} content={message.content} timeAgo={message.timeAgo}/>:
+              <ReceiverMessage key={index} name={message.sender.name} content={message.content} timeAgo={message.timeAgo}/>
```
Comment (Kajol-Kumari) [inline frontend/screens/chat/ChatInterface.js:86]: try to use optional check here so that it doesn't break the feature like `message?.sender?.name`

File: frontend/screens/inbox/InboxInterface.js [hunk 3]
```diff
@@ -25,90 +39,92 @@ const InboxInterface = () => {
...
+            <View style={styles.headerTitleContainer}>
+              <Text style={styles.headerTitle}>Inbox</Text>
+            </View>
+          </TouchableOpacity>
+          <TouchableOpacity onPress={() => navigation.navigate("NewMessage")}>
+            <View style={styles.newMessageButton}>
+              <Image
+                style={styles.newMessageIcon}
+                contentFit="cover"
+                source={require("../../assets/depth-5-frame-021.png")}
```
Comment (Kajol-Kumari) [inline frontend/screens/inbox/InboxInterface.js:72]: the name of the file isn't very informative, can you rename the file so that ut clearly states what's inside it?

File: frontend/screens/inbox/InboxInterface.js [hunk 4]
```diff
@@ -25,90 +39,92 @@ const InboxInterface = () => {
...
-            />
+      </View>
+      <ScrollView onScroll={handleScroll} contentContainerStyle={styles.scrollViewContent}>
+        <View style={[styles.inboxContainer, { height: Dimensions.get("window").height }]}>
+          <View style={styles.searchContainer}>
+            <View style={styles.searchContent}>
+              <Image
+                style={styles.searchIcon}
+                contentFit="cover"
+                source={require("../../assets/depth-3-frame-04.png")}
```
Comment (Kajol-Kumari) [inline frontend/screens/inbox/InboxInterface.js:85]: here also, the name of the file `depth-3-frame-04` isn't very informative, can you rename the file so that ut clearly states what's inside it?

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]

File: frontend/controller/connectionsController.js (hunk 1 above)
Comment (BHS-Harish) [reply frontend/controller/connectionsController.js:211]: I got the latest message on conversation.messages[0] only the older one shows first message.

**Comment by RohitGupta1237** [comment]
@BHS-Harish  make time stamps in hours instead of minutes
